src/test/scala/firrtlTests/SimpleArtihmeticTests.scala

```diff
@@ -0,0 +1,48 @@
+// See LICENSE for license details.
+
+package firrtlTests
+
+import firrtl.transforms.SimpleArithmetic
+
+/**
+ * Tests inline instances transformation
+ */
+class SimpleArtihmeticTests extends LowTransformSpec {
```

Comment:
Add test case for overflow (max integer).

Reply:
@azidar @aswaterman Check that my test that I added is correct. For a 2 bit signal I put -3.

```
  "The module Top" should "not convert sub(_, -<max value>) to add(_, 1)" in {
    val input =
      """circuit Top :
        |  module Top :
        |    input a : SInt<2>
        |    output y : SInt<2>
        |    y <= sub(a, SInt(-3))""".stripMargin
    val check =
      """circuit Top :
        |  module Top :
        |    input a : SInt<2>
        |    output y : SInt<2>
        |    y <= asSInt(bits(sub(a, SInt(-3))), 1, 0))
        """.stripMargin
    execute(input, check, Nil)
  }
```

Reply:
You want `SInt(-2)` here.